packages/asset/contracts/Catalyst.sol

```diff
@@ -16,20 +18,14 @@ contract Catalyst is
     ERC1155Upgradeable,
     ERC1155BurnableUpgradeable,
     ERC1155SupplyUpgradeable,
+    ERC1155URIStorageUpgradeable,
     ERC2771Handler,
-    AccessControlUpgradeable
+    AccessControlUpgradeable,
+    OperatorFiltererUpgradeable
 {
     bytes32 public constant MINTER_ROLE = keccak256("MINTER");
 
-    uint256 public constant COMMON_CATALYST_ID = 1;
-    uint256 public constant UNCOMMON_CATAYST_ID = 2;
-    uint256 public constant RARE_CATALYST_ID = 3;
-    uint256 public constant EPIC_CATALYST_ID = 4;
-    uint256 public constant LEGENDARY_CATALYST_ID = 5;
-    uint256 public constant MYTHIC_CATALYST_ID = 6;
-
-    uint256 public catalystTypeCount = 6;
-
+    uint256 public catalystTierCount;
     address private royaltyRecipient;
     mapping(uint256 => uint256) private catalystRoyaltyBps;
```

Comment:
We should regroup all the royalty stuffs and put it into another contract that this one would extend

```
Catalyst is RoyaltyWhatever {

}

RoyaltyWhatever {
  __RoyaltyWhatever_init() {
    recipient = 
    bps =
  }
}